internal/resource/config/authenticationpolicies/policy/authentication_policies_policy_resource.go

```diff
@@ -0,0 +1,252 @@
+	providerConfig internaltypes.ProviderConfiguration
+	apiClient      *client.APIClient
+}
+
+type authenticationPoliciesPolicyModel struct {
+	Id                              types.String `tfsdk:"id"`
+	PolicyId                        types.String `tfsdk:"policy_id"`
+	Name                            types.String `tfsdk:"name"`
+	Description                     types.String `tfsdk:"description"`
+	AuthenticationApiApplicationRef types.Object `tfsdk:"authentication_api_application_ref"`
+	Enabled                         types.Bool   `tfsdk:"enabled"`
+	RootNode                        types.Object `tfsdk:"root_node"`
+	HandleFailuresLocally           types.Bool   `tfsdk:"handle_failures_locally"`
+}
+
+func (r *authenticationPoliciesPolicyResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	schema := schema.Schema{
+		Description: "Manages an Authentication Policy",
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				Optional:    true,
+				Description: "The authentication policy name. Name is unique.",
+			},
+			"description": schema.StringAttribute{
+				Optional:    true,
+				Description: "A description for the authentication policy.",
+			},
+			"authentication_api_application_ref": resourcelink.CompleteSingleNestedAttribute(true, false, false,
+				"Authentication API Application Id to be used in this policy branch. If the value is not specified, no Authentication API Application will be used."),
+			"enabled": schema.BoolAttribute{
+				Optional:    true,
+				Computed:    true,
+				Default:     booldefault.StaticBool(true),
+				Description: "Whether or not this authentication policy tree is enabled. Default is true.",
+			},
+			"root_node": authenticationpolicytreenode.ToSchema(),
+			"handle_failures_locally": schema.BoolAttribute{
+				Optional:    true,
+				Computed:    true,
+				Default:     booldefault.StaticBool(false),
+				Description: "If a policy ends in failure keep the user local.",
+			},
+		},
+	}
+	id.ToSchema(&schema)
+	id.ToSchemaCustomId(&schema, "policy_id", false, false, "The authentication policy ID. ID is unique.")
+	resp.Schema = schema
+}
+
+// Metadata returns the resource type name.
+func (r *authenticationPoliciesPolicyResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_authentication_policies_policy"
+}
+
+func (r *authenticationPoliciesPolicyResource) Configure(_ context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+
+}
+
+func readAuthenticationPolicyResponse(ctx context.Context, r *client.AuthenticationPolicyTree, state *authenticationPoliciesPolicyModel) diag.Diagnostics {
```

Comment:
In other resources/datasources we have put this method and the model in a separate _common.go file, since it is being used both here and in the data source